src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,498 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+                    validationFailed.size = cockpit.format(_("Storage volume size must not exceed the storage pool's capacity ($0 $1)"), poolCapacity.toFixed(2), diskParams.unit);
+                }
+            } else if (mode === USE_EXISTING) {
+                if (mode !== CUSTOM_PATH && !diskParams.existingVolumeName)
+                    validationFailed.existingVolumeName = _("Please choose a volume");
+            }
+
+            return validationFailed;
+        };
+        return validateParams();
+    }, [diskParams, mode, customDiskVerificationFailed, customDiskVerificationMessage, storagePoolName]);
+    const validationFailed = validate ? _validationFailed : {};
+
+    let defaultBody;
+    const dialogLoading = storagePools === undefined;
+    if (dialogLoading) {
+        defaultBody = (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    } else if (isMediaInsertion) {
+        defaultBody = (
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
+                <FormGroup fieldId={`${idPrefix}-source`}
+                           id={`${idPrefix}-source-group`}
+                           label={_("Source")} isInline hasNoPaddingTop>
+                    <Radio id={`${idPrefix}-custompath`}
+                           name="source"
+                           label={_("Custom path")}
+                           isChecked={mode === CUSTOM_PATH}
+                           onChange={() => onValueChanged('mode', CUSTOM_PATH)} />
+                    <Radio id={`${idPrefix}-useexisting`}
+                           name="source"
+                           label={_("Use existing")}
+                           isChecked={mode === USE_EXISTING}
+                           onChange={() => onValueChanged('mode', USE_EXISTING)} />
+                </FormGroup>
+                {mode === USE_EXISTING && (
+                    <UseExistingDisk idPrefix={`${idPrefix}-existing`}
+                                     onValueChanged={onValueChanged}
+                                     storagePoolName={storagePoolName}
+                                     existingVolumeName={diskParams.existingVolumeName}
+                                     validationFailed={validationFailed}
+                                     vmStoragePools={storagePools}
+                                     vms={vms}
+                                     vm={vm} />
+                )}
+                {mode === CUSTOM_PATH && (
+                    <CustomPath idPrefix={idPrefix}
+                                onValueChanged={onValueChanged}
+                                hideDeviceRow
+                                validationFailed={validationFailed} />
+                )}
+            </Form>
+        );
+    } else {
+        defaultBody = (
+            <>
+                <Form onSubmit={e => e.preventDefault()} isHorizontal>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231117-123758-a6ff1cb9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)